### DISCUSSION TITLE
False power outage alerts

### DISCUSSION THREAD

**Discussion by thomasshawn** [jgyates/genmon#1043] [Questions] [answered]
.

### Expected Behavior

Been flawless for about 2 weeks. Only alerts before now have been appropriately when unit was  exercising or a manual start. 

### Actual Behavior

Multiple times a day receiving false power outage alerts. Started yesterday afternoon right after the weekly exercise run of the generator.  Had 15+ since then. 

### Steps to Reproduce (including precondition)
Many times an hour. Possibly related to dropped packets? Already added time to the addition modbus timeout (2 secs then 3 secs now at 5 secs) and it seemed to help for a few hours now it's doing it again. 

### Your Environment

- Generator Model: briggs
- Genmon Logs: Submitted
- Genmon Version: 1.18.18

**Comment by jgyates** [comment, marked as answer]
These false power losses are known issues caused by communication timeouts that unfortunately are beyond my control. The root issue is that the modbus protocol has some limitations and generac did not compensate for these limitations in the implementation of their firmware, probably because they did not anticipate the modbus link being used in an error prone environment like over a TCP link. These best way to keep this form happening is to ensure that you have zero communication errors. Timeout errors, depending on the reason of the error, will cause this type of problem and you log has a lot of timeout errors in your log that match up with the entries in your outage log.

Number 16 on the known issues list covers this in more detail: https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues

It looks like you are using serial over TCP. If Wifi is involved, make sure that your signal strength is optimal. Timeout errors due to weak wifi signals are more prone to this type of error as the WiFi protocol will repeat sending the packet for a period of time if the packet is not received, which adds delays to the potential reception of a packet that was missed in the initial transmission.. This delay will often make the packet transmitted from the controller to be received by genmon at a time beyond the modbus timeout, which causes genmon to believe received the wrong data for a requested modbus register.

I am going to close this issue but feel free to post to this thread if you have any other questions.

**Comment by thomasshawn** [comment]
Not generac so wasn't sure if it was same issue. I am on wifi but the device sits next to my router and signal strength is quite good. How long can i extend the mod bus timeout before it starts causing its own problems? 

I will experiment with hardwired as well for that bridge and see if that helps as well.

**Comment by jgyates** [comment]
You still have the same issue with other controllers. If you stick to the modbus protocol, you will have this issue with a loss prone medium like wifi. To correct this you would have to break the modbus protocol an add a sequence number at a minimum to the protocol. 

You can extend the timeout for a long time, however genmon will potentially send you and email after 8 seconds of not packet activity, but genmon will still function with the delayed packet timeout.

Wifi, even with a good signal strength, is not perfect. Even with a strong signal you can get interference due to reflections of the signal on non absorbing surfaces. In terms of reliability, wired is always better than wireless.

**Comment by jgyates** [comment]
If you try wired let us know how it goes as I am sure others would find that data point useful. I struggled with this is same issue as my generator is over a 100 feet away and wifi was weak. I eventually dropped a cable and it works a lot better for me.

**Comment by thomasshawn** [comment]
My current setup is about 25 feet away but I ran wire from generator to my  router. The wire is plugged into a wifi bridge sitting right next to my pi which is plugged directly into my network switch.

I have spent hours with the manufacturer's support of the w630 device. (It's identical to the 610) they offer little to no support getting the lan option to work, this morning they basically told me to buy another one of their devices.

I tried multiple of their suggested settings but the language barrier prevented  meaniful troubleshooting, and I also think I may have had multiple agents in the conversation as they seemed to require re-explaination of the desired outcome many times.

So, if anyone has working settings for the 610/630 getting ethernet connected I'd appreciate it.

I'm going to try the usr-tcp-232-306 this afternoon and see if I get anywhere with it. No wifi only ethernet but also doesn't have a ground wire only has Tx/RX

**Comment by jgyates** [comment]
The usr-tcp-232-306 looks like it supports both RS422/485 (on the terminal bock connector) and RS-232 on the (DB-9 connector). My guess is that pin 5 on the DB-9 is ground (which is standard for RS-232). The RS422/RS485 terminal block does not have ground (only TX and RX) because it is not required for RS422/RS485 as it is a differential interface (i.e. the difference between the + and - determine the if it is a one or a zero.

**Comment by jgyates** [comment]
also, if you power the usr-tcp-232-306  via the terminal block to to the 5V and ground on the molex connector the ground is *probably* taken care of. I have powered a USR device like this before.

**Comment by thomasshawn** [comment]
My plan was to power the usr-tcp232-306 by the included powerbrick since I've already run the wires inside the house to be closer to the router and network switch.

Traveling for next few days so it likely will be a few days before I can test it.

**Comment by jgyates** [comment]
then you will need to make sure pin 5 on the DB9 is connected to ground on the molex connector.

**Comment by thomasshawn** [comment]
Don't plan on doing db9. I've run cat 5e from generator to the inside location. Continuity is good no broken wires already checked. On the w630 there are 3 wires rx/tx/fed but the tcp232-306 only has rx/TX. That was why I preferred using the w630 but I can't get the settings on the lan side to work. And tech support wasn't helpful .

**Comment by jgyates** [comment]
Maybe I am missing something. I don't understand your connection. The controller interfaces RS-233, and the usr-tcp232-306 only communicates via RS-232 on the DB-9 connector. The green terminal block labeled T+ and T- is for RS422/RS485 which is not compatible with the controller. I don't understand how you are not planning on using the DB9. That is the only connector on the USR device that will interface to the molex.

**Comment by thomasshawn** [comment]
The controller is rs485.
The bridge uses either 232, or 485.
What little bit of information I did get out out of pusr, they told me it was just easier to use the TCP 232 device than to try and use the ethernet port of the w630 even though they advertise it as being able to do everything, Wi-Fi internet or even both simultaneously.

My plan is not to use the DB9 but to use the connector just as I have it it running Cat5 cable into the green terminal block

**Comment by thomasshawn** [comment]
![16970649538497384073155042904352](https://github.com/jgyates/genmon/assets/118871294/f581ad26-734e-40dd-ac2b-0c43be7d205e)
Current setup

**Comment by jgyates** [comment]
I keep forgetting you are using a Briggs controller, sorry. :)
you will not need the ground.

**Comment by thomasshawn** [comment]
I just cannot get the w630 to function with ethernet. So need to switch to the tcp232 module

**Comment by thomasshawn** [comment]
unplugged everything.  default setting on W630. plugged in network cable before i powered on the W630.
then I accessed the wifi on the W630 in default mode and all the options to run it as a wired AP were already checked. i simply changed my preferred IP address and then rebooted the device and it connected to my network and started sending data. I'll watch logs and send you some in the AM. <holding my breath>
